src/lib/y2storage/boot_requirements_strategies/base.rb

```diff
@@ -92,19 +92,35 @@ def warnings
           res << SetupError.new(message: error_message)
         end
 
+        if boot_in_thin_lvm?
+          error_message =
+            _("The device mounted at '/boot' should not be in a thinly provisioned LVM VG.")
+          res << SetupError.new(message: error_message)
+        end
+
+        if boot_in_bcache?
+          error_message =
+            _("The device mounted at '/boot' should not be in a BCache.")
+          res << SetupError.new(message: error_message)
+        end
+
         res
       end
 
       # All fatal boot errors detected in the setup, for example, when a / partition
       # is missing
       #
+      # @note errors must be raised only in those scenarios making the installation impossible.
+      #   For the rest circumstances please use {#warnings} instead.
```

Comment:
"For the rest circumstances" sounds a bit broken. Maybe "for any other circumstance" or "for the rest of circumstances".

Reply:
Right. At some point I ate the "of" preposition (a general problem that I have with them :stuck_out_tongue_winking_eye:).

Anyway, I'll fix it using your first option.

Thanks!